dupin/detect/dynp.py

```diff
@@ -0,0 +1,46 @@
+import _DynP
+
+class DynP:
+    """Dynamic Programming class for calculating optimal segmentation
+
+    Attributes:
+        data (np.ndarray): Matrix storing the dataset.
+        num_bkps (int): Number of breakpoints to detect.
+        jump (int): Interval for checking potential breakpoints.
+        min_size (int): Minimum size of a segment.
+    """
+
+    def __init__(self, data: np.ndarray, num_bkps: int, jump: int, min_size: int):
+        """Initializes the DynamicProgramming instance with given parameters."""
+        self.dynp = _DynP.DynamicProgramming(data, num_bkps, jump, min_size)
+
+    def set_num_threads(self, num_threads: int):
+        """Sets the number of threads for parallelization.
+
+        Args:
+            num_threads (int): The number of threads to use.
+        """
+        self.dynp.set_threads(num_threads)
+
+    def return_breakpoints(self) -> list:
```

Comment:
This should be unnecessary.

Reply:
can you clarify which part

Reply:
I think you meant return_breakpoints, I removed it and just kept fit.